src/modules/information/recent-history/stake/index.ts

```diff
@@ -0,0 +1,69 @@
+import axios from 'axios';
+import { ethers } from 'ethers';
+import { TOKEN_API_URL } from 'src/modules/token-api';
+import { DappCombinedInfo } from 'src/v2/models';
+import { RecentHistory, RecentHistoryTxType } from './../../index';
+
+interface UserStakeHistory {
+  timestamp: string;
+  contractAddress: string;
+  transaction: string;
+  amount: string;
+  transactionHash?: string;
+  transactionSuccess?: boolean;
+}
+
+export const castStakeTxType = (txType: RecentHistoryTxType): string => {
+  switch (txType) {
+    case 'BondAndStake':
+      return 'Stake';
+    case 'NominationTransfer':
+      return 'Nomination Transfer';
+    case 'UnbondAndUnstake':
+      return 'Unstake';
+    default:
+      return txType;
+  }
+};
+
+export const getStakeTxHistories = async ({
+  address,
+  network,
+  symbol,
+  dapps,
+  subScan,
+}: {
+  address: string;
+  network: string;
+  symbol: string;
+  dapps: DappCombinedInfo[];
+  subScan: string;
+}): Promise<RecentHistory[]> => {
+  const url = `${TOKEN_API_URL}/v1/${network}/dapps-staking/stats/user/${address}/1%20years`;
+  const result = await axios.get<UserStakeHistory[]>(url);
+  const numberOfHistories = 5;
+  return result.data
+    .filter(
+      (it) =>
+        it.transaction === 'BondAndStake' ||
```

Comment:
can we use separate type value instead of hardcoded string?

Reply:
ok, I've defined type annotation in [this](https://github.com/AstarNetwork/astar-apps/pull/602/commits/cca91411a2dbee35ba79a4e5a3ca054d842da6c9) commit.